returnn/tf/layers/rec.py

```diff
@@ -393,6 +393,9 @@ def get_out_data_from_opts(cls, network, unit, sources=(), initial_state=None, *
       out_batch_dim_axis = out.batch_dim_axis  # maybe the subnet does not have a batch-axis (rare but valid)
       deps += subnet.get_parent_deps()
     assert out
+    if out.is_batch_feature_major and out.time_dim_axis is not None:
+      # this correctly reorders the axes template when the input is (batch, feature, time)
+      out = out.copy_as_time_major()
```

Comment:
Why is this needed? We anyway overwrite time-dim-axis and batch-dim-axis below. Maybe just also reset feature-dim-axis, like:
```
out.feature_dim_axis = NotSpecified
```

Reply:
If I do not do this, I will end up having `batch_dim_axis=1` and `feature_dim_axis=1`. Why can I set the feature dim axis to be `NotSpecified`? What will happen then?

Reply:
If it is not specified (which is the default), it will use the default feature dim axis, which is -1 for this case. (See `_default_feature_dim_axis` for the slightly more complicated behavior.)

Reply:
I do not see any way to avoid this, as the `batch_shape` needs to be corrected as well. I could remove the first part of the if though, as we just need to check for the time axis. Just setting the `feature_dim_axis` is definitely not working.

Reply:
Then add whatever is missing.
Using `copy_as_time_major` is anyway not really right, and confusing. Time-major just says that time should be first (i.e. time major), it doesn't say anything about the other dims. E.g. it could still be (T,F,B) or whatever else.
In any case, we know exactly how the output should look like, no matter the input. We should maybe not derive from the input at all (except for the check whether it has a time-dim, which is relevant), and definitely not do fancy things like `copy_as_time_major`, which is really even more ambiguous and unclear. Think about the future reader of this code. If the future reader wants to know, how the output looks like. With this code, it is very unclear, the reader need to think about many different possible cases. But this is totally misleading, because it is very clear, and no matter the input, it will always be the same. There are simply only two possible cases:
(T,B,F) if there is a time-dim, otherwise (B,F). Nothing else matters from the input. The code should make this very clear when looking at it.